bazel-mode.el

```diff
@@ -0,0 +1,240 @@
+(defun sort-strings-longest-first (l)
+  (sort l #'(lambda (s1 s2) (> (length s1) (length s2)))))
+
+;; define several category of keywords
+(setq bazel-keywords
+  '("for"))
+
+(setq android-rules
```

Comment:
I don't think you should enumerate all known rules here. Users can add arbitrary rules (see https://docs.bazel.build/versions/master/skylark/concepts.html), so such a list would always be incomplete.

If you want to add font lock rules for rule names, you could match on an identifier at the beginning of a line followed by an open parenthesis.

But it's probably easiest to not add font locking for rule names, they stand out as-is.

Reply:
Fair enough. I'd like to hear your feedback for a few ideas:

1. Enumerating the Bazel builtin rules (cc_library, e.g.) but not attempting to enumerate those that are added via Skylark rules (go_library, e.g.)? That's not totally satisfactory because some rules would enjoy syntax highlighting but others wouldn't, which is incongruous.

2. Not enumerating any rules for now, but following this pull request up with one that updates bazel-mode to be based on python-mode instead of prog-mode, or otherwise getting the syntax highlighting rules of the sort you mention ("it's a rule name if it starts at the beginning of a line and is punctuated with an open parenthesis", etc.)?

Reply:
If we go with option 1, we should consider a mechanism to let clients of this code configure additional rule names that they'd like bazel-mode to recognize.

Reply:
My preference would be for 3: Don't enumerate any rules, and keep bazel-mode inheriting from prog-mode. I think it's different enough from Python that inheriting from python-mode causes more problems than it solves. (This is only true for BUILD and WORKSPACE files, not .bzl files). If you want, add a font lock rule for rule names so that they look different from e.g. attributes, but even that I think isn't strictly necessary, they are usually easy to recognize without special highlighting.

Reply:
(Just responding since Github has marked this thread as  "outdated")

Reply:
The rules have been removed. We can revisit this in a later PR with an alternative approach to enumeration.

Reply:
Thanks, please also remove the attribute names. The same argument applies for them as well.